lib/Runtime/Library/JavascriptNumber.cpp

```diff
@@ -1032,49 +1034,42 @@ namespace Js
 
         JavascriptString *dblStr = JavascriptString::FromVar(FormatDoubleToString(value, NumberUtilities::FormatFixed, -1, scriptContext));
         const char16* szValue = dblStr->GetSz();
+        const size_t szLength = dblStr->GetLength();
 
-#ifdef ENABLE_GLOBALIZATION
-        count = GetNumberFormatEx(LOCALE_NAME_USER_DEFAULT, 0, szValue, NULL, NULL, 0);
+        pszRes = szRes;
+        count = Numbers::Utility::NumberToDefaultLocaleString(szValue, szLength, pszRes, bufSize);
 
-        if( count <= 0 )
+        if ( count == 0 )
         {
             return dblStr;
         }
         else
         {
-            if( count > bufSize )
+            if ( count > bufSize )
             {
                 pszRes = pszToBeFreed = HeapNewArray(char16, count);
-            }
-            else
-            {
-                pszRes = szRes;
-            }
 
-            int newCount = GetNumberFormatEx(LOCALE_NAME_USER_DEFAULT, 0, szValue, NULL, pszRes, count);
+                count = Numbers::Utility::NumberToDefaultLocaleString(szValue, szLength, pszRes, bufSize);
```

Comment:
I'm not sure if I understood the 4 calls part. Perhaps you missed the if statement above. Agree on replacing bufSize to count and did that.